scipy/stats/anova.py

```diff
@@ -0,0 +1,268 @@
+                                      ci=ConfidenceInterval(low=diff - t*se,
+                                                            high=diff + t*se))
+        return result
+
+
+def anova_oneway(*groups):
+    """
+    One-way (i.e. single factor) analyis of variance.
+
+    Parameters
+    ----------
+    group1, group2, ... : 1-d sequences
+        The groups of values to be tested.  Each group must
+        be a one-dimensional sequence of values.
+
+    Returns
+    -------
+    aov : AnovaOnewayResult
+        An object whose attributes contain the information normally
+        presented in an ANOVA table.  The object also has the methods
+        `means_ci()` that computes the confidence intervals for the
+        means of each group, and `deltas()`, the computes the confidence
+        intervals for the pairwise differences in the means of the groups.
+
+    References
+    ----------
+    .. [1] Jerrold H. Zar, Biostatistical Analysis (fifth edition),
+           Prentice Hall, Upper Saddle River, New Jersey USA (2010)
+
+    Examples
+    --------
+    >>> from scip.stats.anova import anova_oneway
+
+    This is Example 10.1 from Zar [1]_.
+
+    >>> feed1 = [60.8, 67.0, 65.0, 68.6, 61.7]
+    >>> feed2 = [68.7, 67.7, 75.0, 73.3, 71.8]
+    >>> feed3 = [69.6, 77.1, 75.2, 71.5]
+    >>> feed4 = [61.9, 64.2, 63.1, 66.7, 60.3]
+    >>> aov = anova_oneway(feed1, feed2, feed3, feed4)
+
+    When the `aov` object is printed, it is formatted as an
+    ANOVA table.
+
+    >>> print(aov)
+    ANOVA one-way
+    Source                   SS  DF            MS            F           p
+    Between groups 338.93736842   3  112.97912281  12.04040385  0.00028301
+    Within groups  140.75        15  9.38333333
+    Total          479.68736842  18  26.64929825
+
+    The p-value is 0.000283, which suggests that the means of the four
+    populations from which the samples were drawn are not all the same.
+    """
+    num_groups = len(groups)
+    groups = [np.asarray(arg, dtype=np.float64) for arg in groups]
+    means = [group.mean() for group in groups]
+    n = 0
+    grand_total = 0
+    for group in groups:
+        n += len(group)
+        grand_total += group.sum()
+    grand_mean = grand_total / n
+
+    v = sum(((group - grand_mean)**2).sum() for group in groups)
+    vb = sum(len(group)*(group.mean() - grand_mean)**2 for group in groups)
+
+    # Check for the edge case where the values in each group are constant.
+    # When this happens, vw is 0.  If we don't handle this explicitly, vw
+    # might contain numerical noise, and then F will be nonsense.
+    if all([np.all(group[0] == group) for group in groups]):
+        vw = 0.0
+    else:
+        vw = v - vb
+
+    dfb = num_groups - 1
+    dfw = n - num_groups
+    msb = vb / dfb
+    msw = vw / dfw
+    if msw > 0:
+        F = msb / msw
+    else:
+        F = np.inf
+    dof_num = num_groups - 1
+    dof_den = n - num_groups
```

Comment:
Is it not possible to reuse the values of `dfb` and `dfw`? They seem to have the same definitions and I don't see anything intervening that would change them, but maybe I'm missing something.